### PR TITLE
Prow sub: create git client only when token is supplied

### PR DESCRIPTION
**Pull request by chaodaiG**
Fix the bug that crash looping sub when github token is not provided

### CODE REVIEW

File: prow/pubsub/subscriber/subscriber.go [hunk 1]
```diff
@@ -194,13 +194,13 @@ func (prh *presubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pe ProwJobEven
...
-		// Fall back to static presubmits to avoid deadlocking when a presubmit is used to verify
-		// inrepoconfig
-		logrus.WithError(err).Debug("Failed to get presubmits")
-		presubmits = cfg.GetPresubmitsStatic(orgRepo)
+	presubmits := cfg.GetPresubmitsStatic(orgRepo)
+	if prh.GitClient != nil { // Get from inrepoconfig only when GitClient is provided
+		var err error
+		presubmits, err = cfg.GetPresubmits(prh.GitClient, orgRepo, baseSHAGetter, headSHAGetters...)
+		if err != nil {
+			logrus.WithError(err).Debug("Failed to get presubmits")
```
Comment (cjwagner) [inline prow/pubsub/subscriber/subscriber.go:202]: I think we need to either return an error or avoid overwriting `presubmits` to `nil` in the error case.

Comment (chaodaiG) [reply prow/pubsub/subscriber/subscriber.go:201]: done

**Review by cjwagner — APPROVED** [review]

**Pull request merged**